### PR TITLE
CEDS-1213 Correctly map WCO DEC using the new mapping strategy

### PR DESCRIPTION
**Pull request by PauloPortugal**
* Map Declaration.Agent.FunctionCode correctly
* Correctly map WCO BorderMeansTransport 
* Fix WCO Consignment mapping 
* Remove TransportInformation form 
* Fix AEOMutualRecognitionParty mapping

### CODE REVIEW

File: app/services/mapping/declaration/AgentBuilder.scala [hunk 1]
```diff
@@ -29,50 +29,66 @@ object AgentBuilder {
...
+            agentAddress.setPostcodeID(postcode)
+          })
+          agent.setAddress(agentAddress)
+      }
+    )
     agent
   }
+  private def setStatusCode(data: RepresentativeDetails) =
+    data.statusCode
+      .map(value => {
```
Comment (PatrykRudnicki) [inline app/services/mapping/declaration/AgentBuilder.scala]: You don't need to put `{` after `value =>`

Just use

```
.map { value =>
   val ....
}
```

It's in a lot of places, it's not wrong, just to inform you :)

File: app/services/mapping/goodsshipment/consignment/ArrivalTransportMeansBuilder.scala [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import wco.datamodel.wco.dec_dms._2.Declaration.GoodsShipment.Consignment.ArrivalTransportMeans
+import wco.datamodel.wco.declaration_ds.dms._2.ArrivalTransportMeansModeCodeType
+
+object ArrivalTransportMeansBuilder {
+
+  def build(implicit cacheMap: CacheMap): GoodsShipment.Consignment.ArrivalTransportMeans =
+    cacheMap
+      .getEntry[WarehouseIdentification](WarehouseIdentification.formId)
+      .filter(_.inlandModeOfTransportCode.getOrElse("").nonEmpty)
+      .map(transportInformation => {
```
Comment (PatrykRudnicki) [inline app/services/mapping/goodsshipment/consignment/ArrivalTransportMeansBuilder.scala]: Same here about parenthesis

Comment (PauloPortugal) [reply app/services/mapping/goodsshipment/consignment/ArrivalTransportMeansBuilder.scala]: I have addressed those

File: app/services/mapping/declaration/AgentBuilder.scala (hunk 1 above)
Comment (PauloPortugal) [reply app/services/mapping/declaration/AgentBuilder.scala]: I have addressed those

**Pull request merged**